core-build/gulp-core-build-typescript/src/miniTasks/BaseCmdTask.ts

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import * as childProcess from 'child_process';
+
+import { GulpTask } from '@microsoft/gulp-core-build';
+
+export interface ICallCmdOptions {
+  onData: (chunk: Buffer) => void;
+  onError: (chunk: Buffer) => void;
+  onClose: (code: number, hasErrors: boolean, resolve: () => void, reject: (error: Error) => void) => void;
+}
+
+/**
+ * @alpha
+ */
+export abstract class BaseCmdTask<TTaskConfig> extends GulpTask<TTaskConfig> {
+  protected _callCmd(
+    binaryPath: string,
+    cwd: string,
+    args: string[],
+    options?: Partial<ICallCmdOptions>
+  ): Promise<void> {
+    options = options || {};
+    const promiseHandlers: ICallCmdOptions = {} as ICallCmdOptions;
```

Comment:
>const promiseHandlers: ICallCmdOptions = {} as ICallCmdOptions; [](start = 4, length = 63)

You could use traditional inheritance here. You won't need this interface anymore. And it would be a bit more clear.